src/components/Tracker/elements/Encounters/Encounters.tsx

```diff
@@ -96,11 +99,31 @@ const Encounters = React.memo(function Encounters() {
   const handleScroll = (index: number) => {
     listRef.current?.scrollToItem(index + 1, 'center');
   };
+  
+  const navigate = useNavigate();
+
+  const handleOpen = (e: React.MouseEvent<HTMLDivElement, MouseEvent>, index: number) => {
+    e.preventDefault();
+    e.stopPropagation();
+    navigate(`/badgedetail/${selectedGame?.value}/${index}`);
+  };
+
+  const handleKeyPress = (e: React.KeyboardEvent<HTMLDivElement>, index: number) => {
+    if (e.key === 'Enter' || e.key === ' ') {
+      navigate(`/badgedetail/${selectedGame?.value}/${index}`);
+    }
+  };
 
   const renderRow: React.FC<RowProps> = ({ index, style }) => {
     const encounter = filteredEncounters[index];
     const foundPokemon = POKEMAP.get(encounter.pokemon);
     const isGreyedOut = encounter?.status?.value === 2;
+    const isTrainerRoute = encounter.zoneID.some(id => ALL_ZONE_IDS.includes(id));
+    let trainerIndex: number | undefined;
+    
+    if (isTrainerRoute) {
+      trainerIndex = ALL_ZONE_IDS.findIndex(id => encounter.zoneID.includes(id));
```

Comment:
You could remove this lookup if you also use your zoneID as the index in ALL_ZONE_IDS. Small performance increase and less verification code. Good for the soul, up to you if you want to change it.